tests/manager/manager-ctor-read_preference-003.phpt

```diff
@@ -0,0 +1,54 @@
+--TEST--
+MongoDB\Driver\Manager::__construct(): read preference options of the wrong type are ignored
```

Comment:
We may want to change this behavior in the future, but this is how processing of Manager options operates today. We can revise this test if we decide to change this in the future.

I'll note that `mongoc-uri.c` is also quite lax about type errors. For instance, options that are expected to be integers are run through `strtol()` with no error handling.